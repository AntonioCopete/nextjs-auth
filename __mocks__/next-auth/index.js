export const getServerSessionMock = jest.fn();

export const getServerSession = getServerSessionMock;

export default jest.fn();
