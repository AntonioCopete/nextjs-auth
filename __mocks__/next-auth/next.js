const NextAuthMock = jest.fn();

const NextAuth = NextAuthMock;
export default NextAuth;
