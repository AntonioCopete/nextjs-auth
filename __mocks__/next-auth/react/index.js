const signInMock = jest.fn();
const SessionProviderMock = jest.fn();
const useSessionMock = jest.fn();

export const signIn = signInMock;
export const SessionProvider = SessionProviderMock;
export const useSession = useSessionMock;
