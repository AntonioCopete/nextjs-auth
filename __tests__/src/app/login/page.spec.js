import { fireEvent, render, screen } from '@testing-library/react';
import Login from '../../../../src/app/login/page';
import { signIn } from 'next-auth/react';

describe('Login', () => {
	beforeEach(() => {
		signIn.mockReset();
	});
	it('should match snapshot', () => {
		const component = render(<Login title="Login" />);
		expect(component).toMatchSnapshot();
	});

	it('should trigger Google login', () => {
		const component = render(<Login title="Login" />);

		const googleButton = screen.getByText('Login with Google');
		fireEvent.click(googleButton);

		expect(signIn).toHaveBeenCalledTimes(1);
		expect(component).toMatchSnapshot();
	});

	it('should fail on signIn', (done) => {
		signIn.mockRejectedValue(new Error('signIn error'));
		const component = render(<Login />);

		const googleButton = screen.getByText('Login with Google');
		fireEvent.click(googleButton);

		setTimeout(() => {
			expect(signIn).toHaveBeenCalledTimes(1);
			expect(component).toMatchSnapshot();
			done();
		}, 300);
	});
});
