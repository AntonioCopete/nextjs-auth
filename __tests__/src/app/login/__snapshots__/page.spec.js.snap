// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="form"
      >
        <h1>
          Login
        </h1>
        <section
          class="socialContainer"
        >
          <div
            class="socialButton"
          >
            <img
              alt="Google"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <p>
              Login with Google
            </p>
          </div>
          <a
            href="/login/email"
          >
            <div
              class="socialButton"
            >
              <img
                alt="Email"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
              <p>
                Login with Email
              </p>
            </div>
          </a>
        </section>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="form"
    >
      <h1>
        Login
      </h1>
      <section
        class="socialContainer"
      >
        <div
          class="socialButton"
        >
          <img
            alt="Google"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <p>
            Login with Google
          </p>
        </div>
        <a
          href="/login/email"
        >
          <div
            class="socialButton"
          >
            <img
              alt="Email"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <p>
              Login with Email
            </p>
          </div>
        </a>
      </section>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
